en/services/illuminator.md

```diff
@@ -0,0 +1,154 @@
+| <span id="MAV_CMD_ILLUMINATOR_ON_OFF"></span>[MAV_CMD_ILLUMINATOR_ON_OFF](../messages/common.md#MAV_CMD_ILLUMINATOR_ON_OFF)                   | Turns illuminators ON/OFF.                                                                                                                                                                 |
+| <span id="MAV_CMD_DO_ILLUMINATOR_CONFIGURE"></span>[MAV_CMD_DO_ILLUMINATOR_CONFIGURE](../messages/common.md#MAV_CMD_DO_ILLUMINATOR_CONFIGURE) | Configures illuminator settings.                                                                                                                                                           |
+| <span id="ILLUMINATOR_STATUS"></span>[ILLUMINATOR_STATUS](../messages/common.md#ILLUMINATOR_STATUS)                                           | Current status of the illuminator. Recommended to publish this at a regular rate.                                                                                                          |
+| <span id="MAV_TYPE_ILLUMINATOR"></span>[MAV_TYPE_ILLUMINATOR](../messages/minimal.md#MAV_TYPE_ILLUMINATOR)                                    | Type of the component (illuminator).                                                                                                                                                       |
+| <span id="MAV_COMP_ID_ILLUMINATOR"></span>[MAV_COMP_ID_ILLUMINATOR](../messages/minimal.md#MAV_COMP_ID_ILLUMINATOR)                           | ID of the component (illuminator).                                                                                                                                                         |
+
+| Enum                                                                                                               | Description               |
+| ------------------------------------------------------------------------------------------------------------------ | ------------------------- |
+| <span id="ILLUMINATOR_MODE"></span>[ILLUMINATOR_MODE](../messages/common.md#ILLUMINATOR_MODE)                      | Illuminator modes.        |
+| <span id="ILLUMINATOR_ERROR_FLAGS"></span>[ILLUMINATOR_ERROR_FLAGS](../messages/common.md#ILLUMINATOR_ERROR_FLAGS) | Fault/health indications. |
+
+### Illuminator Connection
+
+Illuminators are expected to follow the [Heartbeat/Connection Protocol](https://github.com/mavlink/mavlink-devguide/blob/master/en/services/heartbeat.md) and send a constant flow of heartbeats (nominally at 1Hz). Illuminators are distinguished via their unique component ID: [MAV_COMP_ID_ILLUMINATOR](#MAV_COMP_ID_ILLUMINATOR). Once a heartbeat is received, the drone can then send a [MAV_CMD_REQUEST_MESSAGE ](https://mavlink.io/en/messages/common.html#MAV_CMD_REQUEST_MESSAGE) command to the illuminator to receive information, set settings, or control the illuminator. An example below illustrates how a drone can request the status of the illuminator.

```

Comment:
Historically IDs identified the type of a thing but now code is expected to check the type. ANY component id might in theory be used. In practice though, if you only have one illuminator you should use the allocated component id because that is most likely not used by some other illuminator. 
You should check your code to make sure it does check the type and not rely on the component id for type identity.
```suggestion
Illuminators are expected to follow the [Heartbeat/Connection Protocol](https://github.com/mavlink/mavlink-devguide/blob/master/en/services/heartbeat.md) and send a constant flow of heartbeats (nominally at 1Hz). Illuminators are identified via their type [MAV_TYPE_ILLUMINATOR](#MAV_TYPE_ILLUMINATOR).
Individual illuminators are distinguished via their unique component ID, which by default should be [MAV_COMP_ID_ILLUMINATOR](#MAV_COMP_ID_ILLUMINATOR) (though this is not mandated, and any ID may be used.
Once a heartbeat is received, the drone can then send a [MAV_CMD_REQUEST_MESSAGE ](https://mavlink.io/en/messages/common.html#MAV_CMD_REQUEST_MESSAGE) command to the illuminator to receive information, set settings, or control the illuminator. An example below illustrates how a drone can request the status of the illuminator.
```

Reply:
PS There is a lot of code in various flight stacks that does the wrong thing here ^^^, but you won't be broken if you do it the right way, and that code might be.